plugins/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorTests.java

```diff
@@ -285,6 +290,7 @@ public void testNonExistentWithoutIgnoreMissing() throws Exception {
         throws Exception {
         Map<String, Object> document = new HashMap<>();
         document.put("source_field", getAsBinaryOrBase64(file));
+        document.put("resource_name", file);
```

Comment:
Could you add an override for the `parseDocument` method that adds a `boolean includeResourceName` parameter and only add `resource_name` if that parameter is true?

Reply:
That what I'm going to do, please stay here, I don't want to wait for another week to continue this conversation. I will finish this this night